src/lib/y2partitioner/actions/create_partition_table.rb

```diff
@@ -96,10 +94,99 @@ def run?
         false
       end
 
-      # Return the device name of the disk
-      # @return [String]
-      def disk_name
-        controller.disk_name
+      # If the operation have destructive consequences, ask for confirmation
+      # to the user.
+      #
+      # @return [Boolean] true if the action is safe or the user confirmed
+      #   to accept the consequences
+      def confirm
+        return true if safe?
+
+        if disk.filesystem
+          confirm_with_filesystem
+        else
+          confirm_with_nested_devices
+        end
+      end
+
+      # Whether performing the action can be considered non-destructive, from
+      # the point of view of this action
+      #
+      # Destroying an unused partition table or an unused encryption layer is
+      # considered to be non-destructive in that regard.
+      #
+      # @return [Boolean] true if the disk (that may be encrypted) is completely
+      #   empty and unused or if it only contains an empty partition table
+      def safe?
+        desc = disk.descendants.size
+        return true if desc.zero?
+        return false if desc > 1
```

Comment:
just matter of taste, but I do not like much this desc variable. Something like "descendants_count" is nicer for me.

Reply:
It's just a shortcut for the comparison in the next two lines. Keeping it short was exactly the point. And it's a very local thing.